lib/CL/devices/devices.c

```diff
@@ -75,6 +75,7 @@
 /* the enabled devices */
 static struct _cl_device_id* pocl_devices = NULL;
 unsigned int pocl_num_devices = 0;
+unsigned int pocl_implicit_finish = 0;
```

Comment:
Did this have significant performance impact? If so, it'd be better to optimize pocl_is_option_set() instead?

Reply:
It popped up in cachegrind IIRC.

Reply:
... i don't remember the exact impact, but i can measure again. Yes it's possible `pocl_is_option_set()` would be a better place

Reply:
Somehow difficult to believe that it'd be a significant bottleneck, but yep pocl_is_option_set() should have then a caching in place. Is this implicit finish really needed for something still? I'd consider dropping the whole feature.

Reply:
It's not necessary, AFAICT it was always purely for debugging. I have never used it, and i'm not aware of anyone using it.